### PR TITLE
17 remove all unnecessary functions

### PR DESCRIPTION
**Pull request by radkaag**

### CODE REVIEW

File: src/identity/dataloaders/__tests__/Identity.test.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ describe('Identity dataloader', () => {
     expect(message).toEqual('Undefined access token');
   });
 
-  it('should load user', async () => {
+  it('should loadAll user', async () => {
```
Comment (mrtnzlml) [inline src/identity/dataloaders/__tests__/Identity.test.js]: Come on, please read your changes before you commit them... 😊

File: src/flight/queries/__tests__/AllFlights.locale.test.js [hunk 2]
```diff
@@ -38,17 +38,35 @@ beforeEach(() => {
...
+  ).replyWithData(MexCsCzDataset);
+
+  RestApiMock.onGet(
+    config.restApiEndpoint.allLocations({
+      term: 'Praha',
+      locale: 'cs-CZ',
+    }),
+  ).replyWithData(PragueCsCzDataset);
+
+  // it's a hack, will be fixed here {https://github.com/kiwicom/graphql/issues/141}
```
Comment (lspdv) [inline src/flight/queries/__tests__/AllFlights.locale.test.js]: In previous PR @mrtnzlml asked for deleting this comment after he closed #141 as invalid. Mentioning also here so we don't forget 🎉

File: src/location/dataloaders/Locations.js [hunk 3]
```diff
@@ -32,10 +32,13 @@ export default class LocationSuggestionsDataloader {
 
   /**
    * Returns single set of multiple suggestions for single location key.
-   * If you need to load only one (the first) location for location key
-   * you have to use 'LocationDataLoader.load' function.
+   * If you need to loadAll only one (the first) location for location key
```
Comment (lspdv) [inline src/location/dataloaders/Locations.js]: This `loadall` could be also changed back, please?

File: src/flight/types/outputs/RouteStop.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
 // @flow
 
 import { GraphQLObjectType, GraphQLString } from 'graphql';
-import { GraphQLDateTime } from 'graphql-iso-date';
 import type { GraphQLResolveInfo } from 'graphql';
+import { GraphQLDateTime } from 'graphql-iso-date';
```
Comment (michalsanger) [inline src/flight/types/outputs/RouteStop.js]: Why this change?

File: src/location/dataloaders/Location.js [hunk 5]
```diff
@@ -28,25 +28,13 @@ export default class LocationDataLoader {
    * Returns single location suggestion based on single location key.
    */
   async load(locationKey: string, options: ?Options): Promise<Location> {
```
Comment (michalsanger) [inline src/location/dataloaders/Location.js]: It's written in #17:
> remove load and loadMany from src/location/dataloaders/Location.js

so why is it the method still here?

File: src/location/dataloaders/Location.js [hunk 6]
```diff
@@ -3,17 +3,17 @@
...
 
 export default class LocationDataLoader {
-  locationSuggestionsDataLoader: LocationSuggestions;
+  locationsDataLoader: Locations;
   dataLoader: DataLoader<Object, Location[]>;
 
-  constructor(dataloader: LocationSuggestions) {
-    this.locationSuggestionsDataLoader = dataloader;
+  constructor(dataloader: Locations) {
+    this.locationsDataLoader = dataloader;
```
Comment (michalsanger) [inline src/location/dataloaders/Location.js]: Once you remove the `load` method from here, also is this dependency can be removed

File: src/location/dataloaders/Locations.js [hunk 7]
```diff
@@ -35,7 +35,10 @@ export default class LocationSuggestionsDataloader {
    * If you need to load only one (the first) location for location key
    * you have to use 'LocationDataLoader.load' function.
    */
-  async loadByKey(locationKey: string, options: ?Options): Promise<Location[]> {
+  async loadByTerm(
+    locationKey: string,
```
Comment (michalsanger) [inline src/location/dataloaders/Locations.js]: Rename `locationKey` to `term` please

Comment (michalsanger) [reply src/location/dataloaders/Locations.js]: Why `loadMany` isn't renamed into `loadByTerms` but the task in #17 is marked as done?

File: src/location/dataloaders/__tests__/LocationSuggestions.cacheKeyFn.test.js [hunk 8]
```diff
@@ -1,13 +1,13 @@
 // @flow
```
Comment (michalsanger) [inline src/location/dataloaders/__tests__/LocationSuggestions.cacheKeyFn.test.js]: Rename the file to `Locations.cacheKeyFn.test.js` please

File: src/location/dataloaders/__tests__/LocationSuggestions.test.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 // @flow
```
Comment (michalsanger) [inline src/location/dataloaders/__tests__/LocationSuggestions.test.js]: Rename the file to `Locations.test.js` please

File: src/flight/types/outputs/RouteStop.js (hunk 4 above)
Comment (radkag) [reply src/flight/types/outputs/RouteStop.js]: It happened during merge conflicts resolve. It doesn't matter eventually, why to discuss it then?

File: src/location/dataloaders/Location.js (hunk 5 above)
Comment (radkag) [reply src/location/dataloaders/Location.js]: Because the method is used in `/src/flight/dataloaders/Flight.js`. We already talked this, we agreed we can't remove it.

File: src/location/dataloaders/Locations.js (hunk 7 above)
Comment (radkag) [reply src/location/dataloaders/Locations.js]: I kinda got confused with other removed loadMany in this PR.

File: src/location/dataloaders/Location.js (hunk 5 above)
Comment (mrtnzlml) [reply src/location/dataloaders/Location.js]: It actually doesn't make sense to remove it from my point of view (mostly because I don't understand the expected result of #17 very well). This file is a data-loader and data-loader interface should always have `load` and `loadMany` functions... 😞

File: src/flight/types/outputs/RouteStop.js (hunk 4 above)
Comment (michalsanger) [reply src/flight/types/outputs/RouteStop.js]: Just an OCD about why submitting changes that does not matter :-)

File: src/location/dataloaders/Location.js (hunk 5 above)
Comment (michalsanger) [reply src/location/dataloaders/Location.js]: @radkaag  In `/src/flight/dataloaders/Flight.js` it's only about loading locations by term and picking up first item. So let's use there `locationsDataloader.loadByTerm`.

@mrtnzlml expected result of #17 is mostly code cleanup and better readability. I'd say `load` and `loadMany` methods are OK until you are loading only by one kind of id. But with locations we are loading by id, term, area...

File: src/flight/dataloaders/Flight.js [hunk 10]
```diff
@@ -175,14 +181,16 @@ export default class FlightDataloader {
...
-    const normalizedLocations = locations.map(location => {
+    const normalizedLocations = locations.map(async location => {
       if (location.location === undefined) {
         return Promise.resolve(this._stringifyRadius(location));
       }
       // adjust exact location using backend location API
-      return this.locationDataLoader
-        .load(location.location, options)
-        .then(newLocation => newLocation.locationId); // eslint-disable-line promise/prefer-await-to-then
+      const possibleValues = await this.locationsDataLoader.loadByTerm(
```
Comment (michalsanger) [inline src/flight/dataloaders/Flight.js]: Please us the `.then` syntax, it's important here, enables parallel requests. With `await` you are waiting for each request to be finished one by one.

File: src/flight/dataloaders/Flight.js [hunk 11]
```diff
@@ -175,14 +181,14 @@ export default class FlightDataloader {
...
-    const normalizedLocations = locations.map(location => {
+    const normalizedLocations = locations.map(async location => {
       if (location.location === undefined) {
         return Promise.resolve(this._stringifyRadius(location));
       }
       // adjust exact location using backend location API
-      return this.locationDataLoader
-        .load(location.location, options)
-        .then(newLocation => newLocation.locationId); // eslint-disable-line promise/prefer-await-to-then
+      return await this.locationsDataLoader
```
Comment (michalsanger) [inline src/flight/dataloaders/Flight.js]: remove await

File: src/flight/dataloaders/Flight.js [hunk 12]
```diff
@@ -175,14 +181,14 @@ export default class FlightDataloader {
     locale: null | string,
   ) {
     const options = locale !== null ? { locale } : {};
-    const normalizedLocations = locations.map(location => {
+    const normalizedLocations = locations.map(async location => {
```
Comment (michalsanger) [inline src/flight/dataloaders/Flight.js]: remove async
